Change default ISO image type in cloud envs

osherdp opened the pull request:
PSI installations are still using ``full-iso`` as the default image type. Seems like we only changed the default in the template, which doesn't apply to all installation types.

This changes the default when no ``ISO_IMAGE_TYPE`` has been given in configmap. Operator should remain installing the operand with ``minimal-iso`` as required.

## List all the issues related to this PR

- [ ] New Feature <!-- new functionality -->
- [ ] Enhancement <!-- refactor, code changes, improvement, that won't add new features -->
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD <!-- Notice that changes for Dockerfiles/Jenkinsfiles aren't tested in CI due to a known bug. -->

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [ ] None

## How was this code tested?

<!-- Please, select one or more if needed: -->

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @filanov 
/cc @lalon4 
/cc @gamli75 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

osherdp commented:
/hold
inspecting ZTP installations, to make sure we're not making any change there

osherdp commented:
/unhold

osherdp commented:
/hold

danielerez commented:
@filanov Are we ready to change the default value? or we still want to see the effect of the change in SaaS?

gamli75 commented:
> @filanov Are we ready to change the default value? or we still want to see the effect of the change in SaaS?

If this change will only affect the cloud environment - no need to hold it

danielerez commented:
> > @filanov Are we ready to change the default value? or we still want to see the effect of the change in SaaS?
> 
> If this change will only affect the cloud environment - no need to hold it

This PR would would affect also onprem as it changes the default value of the config directly. I.e. it changes the default value for all envs.

osherdp commented:
> > > @filanov Are we ready to change the default value? or we still want to see the effect of the change in SaaS?
> > 
> > 
> > If this change will only affect the cloud environment - no need to hold it
> 
> This PR would would affect also onprem as it changes the default value of the config directly. I.e. it changes the default value for all envs.

That's right
I think it deserves adding support of it in configmap, and changing it only for PSI

filanov commented:
> > > > @filanov Are we ready to change the default value? or we still want to see the effect of the change in SaaS?
> > > 
> > > 
> > > If this change will only affect the cloud environment - no need to hold it
> > 
> > 
> > This PR would would affect also onprem as it changes the default value of the config directly. I.e. it changes the default value for all envs.
> 
> That's right I think it deserves adding support of it in configmap, and changing it only for PSI

I changed it only for cloud so that we can get the statistics and then if it actually resolve media disconnections issues we can change it to be minimal by default as you did. 
Suggesting to hold until we have the data from the cloud.

osherdp commented:
/test edge-e2e-metal-assisted-onprem

osherdp commented:
> I changed it only for cloud so that we can get the statistics and then if it actually resolve media disconnections issues we can change it to be minimal by default as you did.
> Suggesting to hold until we have the data from the cloud.

I changed the scope to just enable us customizing it wherever we want. Please take another look

danielerez commented:
/lgtm

osherdp commented:
> ISO_IMAGE_TYPE=minimal-iso

yes, you're right
thank you!

osherdp commented:
/test edge-e2e-metal-assisted-onprem

osherdp commented:
/test edge-e2e-metal-assisted-kube-api-late-binding-single-node

osherdp commented:
/unhold
late-binding job is hightly flaky, and either way it doesn't seem to affect it
